### ISSUE TITLE
Bug when using `HasMany.addById` multiple times

### ISSUE THREAD

**Issue by y-lohse** [cozy/cozy-client#358]
In #328 , we decided to reshape the API of `HasMany.addById` (and `removeById`) so that it would automatically save changes after we're done.  
I still think this is desirable and a good API, but unfortunately there is a bug when calling one of these functions multiples times in a row.

We're calling `save` by passing the [`Association.target`](https://github.com/cozy/cozy-client/blob/978bd8bfb55febbfde7d7e7ae1f4dfb33563eb68/packages/cozy-client/src/associations/HasMany.js#L110), which works the first time. But after the save, the document is obviously updated, and gets a new `rev`. 
The second time the function is called, `Association.target` still has the old `rev`, so the save fails.

While `save` returns the server response, we can't straight up use this as new `target` since the original target comes from the [`hydrateDocuments`](https://github.com/cozy/cozy-client/blob/978bd8bfb55febbfde7d7e7ae1f4dfb33563eb68/packages/cozy-client/src/CozyClient.js#L467) process, which pulls document from the store — and these documents may not be exactly the same than the server version (for example in the case of documents with `HasManyFiles`).  
I think *maybe* we could just update the `rev` of `target`? But if the response document somehow has other changes that aren't in `target` we'll run into more problems down the line so I'm not sure that's the way to go…

**Comment by ptbrowne** [comment]
Can you remove the _rev from the document ? I am not sure it is needed.

**Comment by y-lohse** [comment]
Nope, just re-tested it to be sure, `_rev` is definitely needed.

**Comment by y-lohse** [comment]
Yeah I think that might make more sense. We'd end up pulling the document from the store quite often but I don't think that's a big problem.
